#2661 Added error message when self registration fails

IsaakKrut opened the pull request:
### Purpose
fixes #2661. 
UI shows "Internal Server Error" when 500 is returned

![image](https://github.com/apache/streampipes/assets/25233902/f7712843-be2f-40e3-abed-efba46b9265d)

### Remarks 

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

bossenti commented:
Can we put a more user-friendly notification text here? E.g., something like: "User registration failed - please report this to your admin"

IsaakKrut commented:
I updated the error message and moved it to the backend. Please let me know if there is a better way to handle the exception. In the commit before the last one, I added a new exception but don't think that was necessary
![image](https://github.com/apache/streampipes/assets/25233902/98d3a0e0-839b-4921-b92f-6b37334994fa)
